include/swift/Basic/SourceLoc.h

```diff
@@ -214,4 +215,31 @@ class CharSourceRange {
 
 } // end namespace swift
 
+namespace llvm {
+template <typename T> struct DenseMapInfo;
+
+template <> struct DenseMapInfo<swift::SourceRange> {
+  static swift::SourceRange getEmptyKey() { return swift::SourceRange(); }
```

Comment:
I'd suggest changing this to:

SMLoc::getFromPointer(DenseMapInfo<const char *>::getEmptyKey())))

for consistency.